sync/src/block/message/response.rs

```diff
@@ -32,10 +34,23 @@ impl Encodable for ResponseMessage {
                 s.append_list(headers);
             }
             ResponseMessage::Bodies(bodies) => {
-                s.begin_list(bodies.len());
-                bodies.iter().for_each(|body| {
-                    s.append_list(body);
-                });
+                s.begin_list(1);
+
+                let mut uncompressed = {
+                    let mut inner_list = RlpStream::new_list(bodies.len());
+                    bodies.iter().for_each(|body| {
+                        inner_list.append_list(body);
+                    });
+                    inner_list.out()
+                };
+
+                let compressed = {
+                    // TODO: Cache the Encoder object
```

Comment:
Do you have a reason to cache it?

Reply:
The document recommended it.
https://docs.rs/snap/0.2.5/snap/struct.Encoder.html

Reply:
The Encoder uses a big size buffer in heap.
Caching the Encoder object will reduce the buffer allocation/deallocation.

Reply:
Actually, that's what I worried.
If you cache the encoder, the memory will not be deallocated. It can make the extension hold very large unused memory.